### ISSUE TITLE
Image do not load - Random issue

### ISSUE THREAD

**Issue by FabriceFroehlyNiji** [SDWebImage/SDWebImage#2768]
Hello,

I have had issues with images for long time now.
Basically, the images work fine. However, every now and then (really hard to debug since it happens almost never) EVERY images of my view just don't load.
There are two tableviews on the view, and scrolling does not fix the rest of the images.
I generally have to switch view or relaunch the app to images back to normal.

I wonder if this issue could not be happening when the library is emptying cache ? I don't know.

I you guys have any idea, i'd really appreciate it.
Btw, I've updated the library to its last version, but that's didn't solve my issue.
Cheers

**Comment by dreampiggy** [comment]
@FabriceFroehlyNiji Please at least fill the issue template. Or we didn't know what your SDWebmage, how does it get integrarted, and the target platform firmware version.

### New Issue Checklist

* [ ] I have read and understood the [CONTRIBUTING guide](https://github.com/rs/SDWebImage/blob/master/.github/CONTRIBUTING.md)
* [ ] I have read the [Documentation](http://cocoadocs.org/docsets/SDWebImage/)
* [ ] I have searched for a similar issue in the [project](https://github.com/rs/SDWebImage/issues) and found none

### Issue Info

 Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform Name           | e.g. ios / macos / tvos / watchos
 Platform Version        | e.g. 12.0 / 10.14.0 / 12.0 / 5.0
 SDWebImage Version      | e.g. 5.0.0 / 4.4.0
 Integration Method      | e.g. carthage / cocoapods / manually
 Xcode Version           | e.g. Xcode 10 / Xcode 9
 Repro rate              | e.g. all the time (100%) / sometimes x% / only once
 Repro with our demo prj | e.g. does it happen with our demo project?
 Demo project link       | e.g. link to a demo project that highlights the issue

### Issue Description and Steps

Please fill in the detailed description of the issue (full output of any stack trace, compiler error, ...) and the steps to reproduce the issue.

**Comment by dreampiggy** [comment]
For this issue. I guess this may related to some cases such as memory warning which cause the cache to be purged.

Is that even the network image can not be downloaded ? Or only the local cached images ? Provide more information can help us to investigate the issue.

**Comment by FabriceFroehlyNiji** [comment]
Info | Value
-- | --
Platform Name | ios
Platform Version | 12.0
SDWebImage Version | 5.0.0
Integration Method | cocoapods
Xcode Version | Xcode 10.2
Repro rate | sometimes 1%
Repro with our demo prj | I can't reproduce it, which makes it difficult to debug. Only testers are having the issue from time to time, randomly, this is a rare issue though

**Comment by FabriceFroehlyNiji** [comment]
Hello @dreampiggy, it is possible that this is due to memory warning. But if the cache is purged, shouldn't the lib download the URLs again instead of looking into cache ?

Images come from URL, they are not on the device. They are downloaded. They also are likely to change (the server can change the image but keep the URL).
Also, it does not come from server because the Android app doesn't have this issue.

**Comment by dreampiggy** [comment]
> But if the cache is purged, shouldn't the lib download the URLs again instead of looking into cache
It will definitely do what you think of.

> EVERY images of my view just don't load.
Is this issue be solved after you switch back another ViewController ? If so, maybe something related to `imageView` level configuration get wrong. It's not something about image downloader or decoder. (Or even you switch back, your image get never get loaded, right ?)

**Comment by FabriceFroehlyNiji** [comment]
> Is this issue be solved after you switch back another ViewController ?

I tried going into a child view and then going back into the parent view, and the images were still not there. I'm pretty sure I need to restart to app for it to work again.

**Comment by dreampiggy** [comment]
> I'm pretty sure I need to restart to app for it to work again.

Is this the black list issue ? Don't you pass `SDWebImageRetryFailed` option ? Or do you have a custom delegate method implements `imageManager:shouldBlockFailedURL:withError:` ?

By default, if you don't pass this option, the HTTP error such as 403, will block the failed url, there are current no API to clear the black list.

**Comment by FabriceFroehlyNiji** [comment]
I am currently not using `SDWebImageRetryFailed ` anywhere. Should I be using it ?

**Comment by dreampiggy** [comment]
@FabriceFroehlyNiji If you want to avoid this black list globally, just set a simple delegate to manager, override `imageManager:shouldBlockFailedURL:withError:` and always return NO.

If you want to black list some of request, you use the option, which asoociated to single image request.

From v5.1.0, you can also use the `option processor`, to have a complicated logic base on image request context and provide the final option, see more in #2736

**Comment by FabriceFroehlyNiji** [comment]
Alright, thank you !
I cannot test right now, but I'll close this and come back if I have issues with the solution you mentioned.

**Comment by FabriceFroehlyNiji** [comment]
FYI, I've added:
``` swift
     func imageManager(_ imageManager: SDWebImageManager,
                      shouldDownloadImageFor imageURL: URL) -> Bool {
        return true
    }
    
    func imageManager(_ imageManager: SDWebImageManager,
                      shouldBlockFailedURL imageURL: URL,
                      withError error: Error) -> Bool {
        return false
    }
```

I'll leave another message if the issue comes back, or if it seems stable.
It'll probably be in more than a month since we don't replicate it very often.

**Comment by FabriceFroehlyNiji** [comment]
It's me again :D
I'm came back to tell you that it seems the solution worked !
Thanks for your help you good lad 💪
